#112311313 Create and manage projects on the dashboard

lewisemm opened the pull request:
### What does this PR do?
-  This PR adds functionality to manage projects i.e. functionality to create, edit, display and delete projects
### Description of task to be completed?
-  Create the functionality to manage projects on the dashboard
### How should this be manually tested?
-  Creating, editing and deleting a personal project
-  Creating, editing and deleting an organisational project
### Any background context you want to provide?
-  The project viewset will create a personal project when an owner is not explicitly specified (in this case, the owner is derived from the request object).
-  In light of this, when creating an organisational project, the front end requires the user to explicitly specify a project owner (an organisation). If no organisation is selected, the app will notify the user of this through a message and the organisation will not be created.
-  User will have to be an admin in an organisation in order to create an organisational project
### What are the relevant pivotal tracker stories
-  The story that corresponds to this PR is #112311313 (Creating and managing projects on dashboard)

collinmutembei committed:
[fix] Update user model

- Update user model to have choices for the user level of a
  member to an organization
- Delete stuff that is not needed, especialy angular-materialize
- Update migrations

collinmutembei committed:
[Feature #112311313] Update projects controller

- Make changes to the projects controller to consume the
  corresponding endpoints for projects
- Update the service to fetch personal, organizational and
  other projects

Commit:
[Fix #111951001] Fix navigation on small devices

Fix display of navbar links on medium and small devices

collinmutembei committed:
[Feature #112311313] Add section for other projects

- Implement a section to display projects that are not
  organizational or personal, i.e personal projects to
  other users that the user has been added to the team
- Add min.map files that had a 404 status code

Commit:
[Feature #112494273] Add placeholders for adding projects

Add a placeholder with a link to create new projects
Add a link to show all projects

Commit:
[Feature #112311313] Add modal for creating projects

Add project creation form

lewisemm committed:
[Feature #112311313] Edit ProjectViewSet to create personal projects

- Retrieves the user id from the request object.
- Intended for use to create only PERSONAL projects

lewisemm committed:
[Feature #112311313] Create logic to add personal projects

- Create a Projects resource in the mainService
- Add a createPersonalProject method in the ProjectCtrl that calls
mainService.createPersonalProject to create a personal project
- Abstract loading of the call to load personal projects to
loadPersonalProjects() function so as to use it to reload personal
projects when a new one is created

lewisemm committed:
[Feature #112311313] Add behaviour to create personal projects

- Wire up the view buttons with models and controller methods to
connect backend behaviour with the front end

lewisemm committed:
[Feature #112311313] Create OrgAssociationViewSet

- Create a viewset to display organisation details for organisations
in which the current user belongs and is an admin member. This endpoint
 is used while creating an organisational project (To populate relevant
organisation names when creating an organisational project)
- Also replaced indentation spaces with tabs

lewisemm committed:
[Feature #112311313] Add org_id read only field to OrgSerializer

- Add org_id field to avail organisational id information to the
view when specifying the organisation to which a project being created
belongs

lewisemm committed:
[Feature #112311313] Add a route for the OrgAssociationsViewSet

- Style the file and replace space indentations with tabs

lewisemm committed:
[Feature #112311313] Create logic to add organisational projects

- Define method createOrganisationalProject
- Abstract logic to fetch organisational projects to a method so as to
call it and refresh organisational projects when a new project is
created

lewisemm committed:
[Feature #112311313] Create resource OrgAssociations in mainService

- Rename method createPersonalProject of Projects resource to
createProject

lewisemm committed:
[Feature #112311313] Wire view controls with behaviour

- Wire up controls to create organisational projects in the view to
respective behaviours in the controller

lewisemm committed:
[Feature #112311313] Define logic to edit organisational projects

- Define read only serializer field 'project_id' for reference when
editing projects
- Define method in the project controller to open a modal dialog to
edit organisational projects and another method to send the edited
data to the back-end restful api
- Update the mainservice.Projects resource to handle the PUT method

lewisemm committed:
[Feature #112311313] Refine Project view and add behaviour

- Display projects as materialize cards
- Define Angular logic to truncate names and descriptions when
they exceed 13 characters
- Use word wrap and edit materialize card margins to give the view
a consistent grid like display of projects

lewisemm committed:
[Feature #112311313] Define logic to edit personal projects

- Refactor controller and view to allow sharing of logic while editing
 both personal and organisational projects

lewisemm committed:
[Finishes #112311313] Define delete logic

- Define method in project controller to open delete modal when delete
button on the view is clicked
- Require confirmation from user before delete operation is started.
User is required to type DELETE (in caps) for the delete operation to
start
- Include delete behaivour in the projects view

andela-anandaa commented on public/static/styles/style.css:
```diff
@@ -156,3 +156,44 @@ h5 {
+    background: #e0e0e0;
+    min-height: 110px;
+    box-shadow: 0 2px 5px 0 #9e9e9e,0 2px 10px 0 #9e9e9e;
+}
```
Separate with blank lines.

andela-anandaa commented on public/templates/limber/login.html:
```diff
@@ -14,14 +14,8 @@ <h4 class="center"><a href="{% url 'index' %}"> Limber </a></h4>
-                            </div>
-
                             <div id="login" class="col s12" >
+                            <span class="active blue-text center" ><h5> LOGIN </h5></span>
```
This is not a semantic mark-up.

andela-anandaa commented on public/templates/limber/projects.html:
```diff
@@ -3,65 +3,247 @@
-</div>
-</div>
-</div>
+<body class="grey lighten-5" ng-controller="ProjectCtrl" ng-cloak>
```
Fix the indentation.

andela-anandaa commented on public/templates/partials/layout.html:
```diff
@@ -40,6 +41,7 @@
          <script>
          $(document).ready(function(){
             $('ul.tabs').tabs();
+            $(".button-collapse").sideNav();
```
All JS should be separated in a `.js` file.

lewisemm committed:
[Fix #112311313] Resolve pull request nitpicks

- Move script tag javascript to app.js
- Fix indentation on project.html
- Fix spacing between selectors on style.css
- Fix html malpractice (remove h5 from span)
